url9a01f43c15_poragchowdhury_securitygame_tgz-pJ8-securitygame_GameMasterJ8/src/securitygame/RationalDefender.java

```diff
@@ -0,0 +1,79 @@
+package securitygame;
+
+import java.util.Random;
+
+/**
+ * BenchMark Defender Agent that takes rational decision based on the defender action cost
+ * @author Porag Chowdhury, Anjan Basak
+ */
+
+public class RationalDefender extends Defender{
+	
+    public RationalDefender(String graphFile)
+    {
+        super("RationalDefender",graphFile);
+    }
+	
+	public void makeMoves()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code blocks on lines 20-46, lines 48-76 into separate methods.